mod/discussions/start.php

```diff
@@ -0,0 +1,582 @@
+<?php
+/**
+ * Discussions
+ */
+
+elgg_register_event_handler('init', 'system', 'discussion_init');
+
+/**
+ * Initialize the discussion component
+ */
+function discussion_init() {
+
+	elgg_register_library('elgg:discussion', elgg_get_plugins_path() . 'discussions/lib/discussion.php');
+
+	elgg_extend_view('css/elgg', 'css/discussion');
+
+	elgg_register_page_handler('discussion', 'discussion_page_handler');
+
+	elgg_register_plugin_hook_handler('entity:url', 'object', 'discussion_set_topic_url');
+
+	// commenting not allowed on discussion topics (use a different annotation)
+	elgg_register_plugin_hook_handler('permissions_check:comment', 'object', 'discussion_comment_override');
+	elgg_register_plugin_hook_handler('permissions_check', 'object', 'discussion_can_edit_reply');
+
+	// discussion reply menu
+	elgg_register_plugin_hook_handler('register', 'menu:entity', 'discussion_reply_menu_setup');
+
+	// allow non-owners to add replies to discussion
+	elgg_register_plugin_hook_handler('container_permissions_check', 'object', 'discussion_reply_container_permissions_override');
+
+	elgg_register_event_handler('update:after', 'object', 'discussion_update_reply_access_ids');
+
+	$action_base = elgg_get_plugins_path() . 'discussions/actions/discussion';
+	elgg_register_action('discussion/save', "$action_base/save.php");
+	elgg_register_action('discussion/delete', "$action_base/delete.php");
+	elgg_register_action('discussion/reply/save', "$action_base/reply/save.php");
+	elgg_register_action('discussion/reply/delete', "$action_base/reply/delete.php");
+
+	// add link to owner block
+	elgg_register_plugin_hook_handler('register', 'menu:owner_block', 'discussion_owner_block_menu');
+
+	// Register for search.
+	elgg_register_entity_type('object', 'discussion');
+	elgg_register_plugin_hook_handler('search', 'object:discussion', 'discussion_search_discussion');
+
+	// because replies are not comments, need of our menu item
+	elgg_register_plugin_hook_handler('register', 'menu:river', 'discussion_add_to_river_menu');
+
+	// add the forum tool option
+	add_group_tool_option('forum', elgg_echo('groups:enableforum'), true);
+	elgg_extend_view('groups/tool_latest', 'discussion/group_module');
+
+	$discussion_js_path = elgg_get_site_url() . 'mod/discussions/views/default/js/discussion/';
+	elgg_register_js('elgg.discussion', $discussion_js_path . 'discussion.js');
```

Comment:
Always put views through simplecache, not directly accessing them through the file system.